### PR TITLE
Option to disable the compositor for better performance

### PR DESCRIPTION
**Pull request by nastys**
Improves performance and reduces stuttering.
Supported window managers:

- KWin/KDE
- Marco/MATE
- XFWM/XFCE.

Enabled by default.

Implements #385

### CODE REVIEW

File: lutris/game.py [hunk 1]
```diff
@@ -320,6 +322,21 @@ def do_play(self, prelaunched, _error=None):
             show_obnoxious_process_monitor_message()
         process_watch = not monitoring_disabled
 
+        if self.runner.system_config.get('disable_compositor'):
+            if os.environ.get('DESKTOP_SESSION') == "plasma":
```
Comment (strycore) [inline lutris/game.py]: Could the block below be implemented in it's own method?

File: lutris/game.py [hunk 2]
```diff
@@ -403,6 +420,9 @@ def on_game_quit(self):
            or self.runner.system_config.get('reset_desktop'):
             display.change_resolution(self.original_outputs)
 
+        if self.compositor_disabled:
+            subprocess.Popen(self.start_compositor, shell=True)
```
Comment (strycore) [inline lutris/game.py]: use `lutris.system.execute` instead of Popen calls

File: lutris/sysoptions.py [hunk 3]
```diff
@@ -205,6 +205,14 @@ def get_dri_prime():
         'advanced': True,
         'help': "Disables process monitoring of games, Lutris won't detect when game quits."
     },
+    {
+        'option': 'disable_compositor',
+        'label': "Disable desktop effects",
+        'type': 'bool',
+        'default': True,
```
Comment (strycore) [inline lutris/sysoptions.py]: Default to False, this is the current behavior.

File: lutris/game.py (hunk 1 above)
Comment (nastys) [reply lutris/game.py]: Done.

File: lutris/game.py (hunk 2 above)
Comment (nastys) [reply lutris/game.py]: Done. I had to allow system.execute to optionally run the command in a shell (default is still False, so it shouldn't break compatibility).

File: lutris/sysoptions.py (hunk 3 above)
Comment (nastys) [reply lutris/sysoptions.py]: I've set it to False, though I'm not sure why it shouldn't be enabled by default.

**Comment by nastys** [comment]
Everything done. By the way, you should be able to commit directly to my fork if you want to make any more edits.

**Comment by strycore** [comment]
Looks all good!
I'm merging it to master after the 0.4.18 release so we can test it more, it will be included in the upcoming 0.5.0 release

**Comment by daniel-j** [comment]
Would this enable `xfwm4` compositor even though I have disabled it before launching the game? I use a custom compton compositor!

**Comment by nastys** [comment]
@daniel-j No, it won't resume xfwm4 if it wasn't previously started.
Unfortunately it doesn't support Compton yet, so it won't make any difference if you enable this option.

**Comment by yaomtc** [comment]
Does this support Compiz?

**Comment by nastys** [comment]
@yaomtc No, it only supports the window managers on the list. I don't think supporting Compiz would be possible, at least not without replacing it with another window manager temporarily (but that may cause some issues).

**Pull request merged**